addons/website_sale/controllers/main.py

```diff
@@ -1252,7 +1252,7 @@ def process_express_checkout(
         # assigned to ensure the right behavior from `shop_payment_confirmation()`.
         request.session['sale_last_order_id'] = order_sudo.id
 
-        if shipping_address and shipping_option:
+        if shipping_address:
```

Comment:
Hi, I just remembered that we had conversation about it with @chevalierv and since it's a test, we wanted to have shipping part showed even when there is no carrier defined.

Reply:
Looks strange to me, even if it's a demo provider, the flows should work the same way as other provider IMHO. We don't want to explain every user that: "we know that you could see the shipping part with the demo provider on this exact same setup, but it doesn't work like that in a real setup. It's an exception." or "Yes, we know it makes no sense to see that part, it's just for the example" :thinking: 

But that's your call in the end =]

Reply:
Ok, github problem, this comment https://github.com/odoo/odoo/pull/113644#discussion_r1331208743 should have been here xD